src/modules/kvs-watch/kvs-watch.c

```diff
@@ -1023,6 +1043,25 @@ static void watcher_cancel (struct ns_monitor *nsm,
     watcher_respond (nsm, w);
 }
 
+static int matchtag_key (flux_t *h,
+                         const flux_msg_t *msg,
+                         char *buf,
+                         size_t bufsize)
+{
+    uint32_t matchtag;
+    const char *uuid;
+    if (flux_msg_unpack (msg, "{s:i}", "matchtag", &matchtag) < 0) {
+        flux_log_error (h, "failed to get matchtag from cancel request");
+        return -1;
+    }
+    if (!(uuid = flux_msg_route_first (msg))) {
+        flux_log_error (h, "failed to get uuid from cancel request");
+        return -1;
+    }
+    snprintf (buf, bufsize, "%s:%u", uuid, matchtag);
+    return 0;
```

Comment:
return result of `snprintf` or check return code?

Reply:
do you mean check the return code of snprintf?  We normally don't b/c the buffer can't be overflowed.   Perhaps there was just a subtlety of your comment I'm missing.  Or perhaps I need a comment indicating that the buffer size is more than sufficient?

Reply:
Well in this case the size is passed into the function, so there's no real way to know that the buffer can't be overflowed unless you check? However, there's only one caller so a comment would suffice (and maybe cast return code to `(void)` to show it is intentional)

Reply:
ahh i get it, it's b/c I moved this code into a function (which it wasn't in a function earlier).  I'll check for overflow.